tests/integration/p2p.go

```diff
@@ -125,12 +129,35 @@ func reconnectPeers(s *state.State) {
 				corelog.Any("Source", sourceAddresses),
 				corelog.Any("Target", targetAddresses))
 
-			err = sourceNode.Connect(ctx, targetAddresses)
+			err = connectWithRetry(ctx, sourceNode, targetAddresses)
 			require.NoError(s.T, err)
 		}
 	}
 }
 
+// connectWithRetry attempts to connect to target addresses with retry logic
+// to handle transient "dial backoff" errors that can occur when libp2p's host
+// is not fully ready to accept connections.
+func connectWithRetry(ctx context.Context, node *state.NodeState, targetAddresses []string) error {
```

Comment:
thought: An alternative to this would be to pass an option to `go-p2p` when testing where it would retry immediately on dial-backoff with a backoff clearing.
```go
// in go-p2p host.go
p.host.Network().(*swarm.Swarm).Backoff().Clear(addrInfo.ID)
```